### ISSUE TITLE
`tf.compat.v1.nn.embedding_lookup` computes wrong forward gradient when indices are out of bound

### ISSUE THREAD

**Issue by VictoriaGriffith** [tensorflow/tensorflow#57902]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9.1

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Current Behaviour?

`tf.compat.v1.nn.embedding_lookup` computes wrong forward gradient when indices are out of bound. In the example code below, `ids=[4]` is out of bound, so the output is all zero. The jacobian therefore is all zeros. However, the forward gradient is wrongly computed as`1.0` w.r.t. to `params[1,0]`.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

params = tf.constant([[0.4], [0.6]], dtype=tf.float32)
ids = tf.constant([4], dtype=tf.int32)

def f(params):
    partition_strategy = "mod"
    validate_indices = False
    max_norm = None
    return tf.compat.v1.nn.embedding_lookup(params, ids, 
    partition_strategy=partition_strategy, 
    validate_indices=validate_indices, 
    max_norm=max_norm, )

with tf.GradientTape(persistent=True) as tape:
    tape.watch(params)
    out = f(params)
print("input params: \n", params)
print("outputs: ", out)


jac = tape.jacobian(out, [params])
print("jacobian: ", jac)

tangents = tf.constant([[0.], [1.]])
with tf.autodiff.ForwardAccumulator(params,tangents) as acc:
  out = f(params)
print("forward outputs: ", out)
jvp = acc.jvp(params)
print("forward gradient: ", jvp)

```

### Relevant log output

```shell
WARNING:tensorflow:Converting IndexedSlices(indices=Tensor("gradient_tape/Reshape_2:0", shape=(1,), dtype=int32), values=Tensor("gradient_tape/Reshape_1:0", shape=(1, 1), dtype=float32), dense_shape=Tensor("gradient_tape/Cast:0", shape=(2,), dtype=int32, device=/job:localhost/replica:0/task:0/device:GPU:0)) to a dense representation may make it slow. Alternatively, output the indices and values of the IndexedSlices separately, and handle the vectorized outputs directly.
input params: 
 tf.Tensor(
[[0.42328522]
 [0.47058594]], shape=(2, 1), dtype=float32)
outputs:  tf.Tensor([[0.]], shape=(1, 1), dtype=float32)
jacobian:  [<tf.Tensor: shape=(1, 1, 2, 1), dtype=float32, numpy=
array([[[[0.],
         [0.]]]], dtype=float32)>]
forward outputs:  tf.Tensor([[0.]], shape=(1, 1), dtype=float32)
forward gradient:  tf.Tensor(
[[0.]
 [1.]], shape=(2, 1), dtype=float32)
```
</details>

**Comment by tiruk007** [comment]
@VictoriaGriffith 
I tried to reproduce the issue on Colab using TF v2.9.1 but getting a different error. Could you please find the gist [here](https://colab.research.google.com/gist/tiruk007/af4bb462f0144c930211fef869a194c8/57902.ipynb) and confirm the issue?

Thank you!

**Comment by VictoriaGriffith** [comment]
@tiruk007 sorry that I forgot to mention this bug is on GPU only.
I turn on GPU in Colab and can reproduce the issue. Could you please confirm the same?

**Comment by tiruk007** [comment]
@sachinprasadhs 
I was able to reproduce the issue in TF v2.10.Please find gist [here](https://colab.research.google.com/gist/tiruk007/bba562c7eb0171a36c32857cd59d3bdd/57902.ipynb)for reference.
Thank you!

**Comment by sachinprasadhs** [comment]
`compat.v1` API's are no longer under support, could you please try `tf.nn.embedding_lookup` and let us know if you observe the similar behavior. Thanks!
